inject/src/main/java/io/micronaut/context/DefaultBeanContext.java

```diff
@@ -1029,8 +1034,15 @@ public <T> T createBean(@NonNull Class<T> beanType, @Nullable Qualifier<T> quali
         ArgumentUtils.requireNonNull("beanType", beanType);
         Optional<BeanDefinition<T>> candidate = findBeanDefinition(Argument.of(beanType), qualifier);
         if (candidate.isPresent()) {
-            try (BeanResolutionContext resolutionContext = newResolutionContext(candidate.get(), null)) {
-                return doCreateBean(resolutionContext, candidate.get(), qualifier, argumentValues);
+            BeanDefinition<T> beanDefinition = candidate.get();
+            try (BeanResolutionContext resolutionContext = newResolutionContext(beanDefinition, null)) {
+                T bean;
+                if (beanDefinition instanceof InstantiatableBeanDefinition<T> instantiatableBeanDefinition) {
+                    bean = resolveByBeanFactory(resolutionContext, instantiatableBeanDefinition, qualifier, argumentValues);
```

Comment:
Don't create a new out-scope variable, just call `return postBeanCreated` right after. Next else is not needed. Same pattern repeats a few times